### PR TITLE
[SPARK-10108] Add since tags to mllib.feature

### PR DESCRIPTION
**Pull request by MechCoder**

### CODE REVIEW

**Comment by MechCoder** [comment]
@mengxr

**Comment by mengxr** [comment]
@MechCoder Shall we use `@Since` annotation directly?

**Comment by MechCoder** [comment]
I thought we can merge all directly and replace everything at one go.

**Comment by mengxr** [comment]
No, for existing PRs, we ask the authors to stay with `@since` tags because most of them are first-time contributors. For new PRs, we should use `@Since` annotation directly.

**Comment by MechCoder** [comment]
Great. Will update in some time.

**Comment by MechCoder** [comment]
@mengxr updated.

I wrote a small bash script that would do the conversion automatically, but there is one corner case that Ihad to take care of maually

**Comment by mengxr** [comment]
@MechCoder Please also check the following:

1. overridden methods that do not require JavaDoc (if the JavaDoc is the same as its parent)
2. public variables in constructors (see examples in https://issues.apache.org/jira/browse/SPARK-9864)

File: mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala [hunk 1]
```diff
@@ -31,6 +31,7 @@ import org.apache.spark.rdd.RDD
  *
  * @param selectedFeatures list of indices to select (filter). Must be ordered asc
  */
+@Since("1.3.0")
 @Experimental
 class ChiSqSelectorModel (val selectedFeatures: Array[Int]) extends VectorTransformer {
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala]: For example, we also need to add annotation to `selectedFeatures`.

**Comment by MechCoder** [comment]
@mengxr I just updated the annotation for one class variable (ChiSqSelector) but it does not show up in the scala doc for variables. (I have attached a screesshot)

screenshot from 2015-08-20 23 12 32

**Comment by MechCoder** [comment]
Actually the annotations on any class variable do not show up in the doc. Can you please verify?

**Comment by mengxr** [comment]
Good catch! I will take a look.

**Comment by mengxr** [comment]
Sent a patch https://github.com/apache/spark/pull/8344 to fix it.

**Comment by mengxr** [comment]
Merged into master and branch-1.5. Thanks! Please make a PR that convert all existing @since tags into @Since annotation (but do not make additional changes to simplify the code review). Then we can have another PR to fix the missing ones.

**Comment by MechCoder** [comment]
Thanks you for the patch.

**Pull request closed without merging**